examples/classifcation/text_classification.exs

```diff
@@ -0,0 +1,273 @@
+# Based on https://valueml.com/named-entity-recognition-using-lstm-in-keras/
+
+# References
+# http://alexminnaar.com/2019/08/22/ner-rnns-tensorflow.html
+
+# Test Data: https://github.com/bhuvanakundumani/NER_tensorflow2.2.0/blob/master/data/test.txt
+
+# Train Data: https://github.com/bhuvanakundumani/NER_tensorflow2.2.0/blob/master/data/train.txt
+
+# Validation Data: https://github.com/bhuvanakundumani/NER_tensorflow2.2.0/blob/master/data/valid.txt
+
+Mix.install([
+  {:axon, github: "elixir-nx/axon"},
+  {:nx, "~> 0.2.1"},
+  {:exla, "~> 0.2.2"}
+])
+
+EXLA.set_as_nx_default([:tpu, :cuda, :rocm, :host])
+
+defmodule TextClassification do
+  require Axon
+  # From Example http://alexminnaar.com/2019/08/22/ner-rnns-tensorflow.html
+  EXLA.set_as_nx_default([:tpu, :cuda, :rocl, :host]) # My :host option doesnt work, jit_apply is undefined
+  require Logger
+  @sequence_length 75
+  @batch_size 128
+  @lstm 128
+  @embed_dimension 256
+  @def_split " "
+  @start "-DOCSTART- -X- -X- O\n"
+  @ending "-------------------------------------------------------------"
+  @incorrect [".", "?", "%", ",", ":", ";", "-docstart-"]
+
+  # seqlen
+  # word_count
+
+  def train(model, data) do
+    model
+    |> Axon.Loop.trainer(&kl_divergence/2, Axon.Optimizers.adam(0.001))
+    |> Axon.Loop.metric(:accuracy, "Accuracy")
+    |> Axon.Loop.run(data, %{}, epochs: 3, iterations: 5)
+
+  end
+
+  defp kl_divergence(y_true, y_pred) do
+    Axon.Losses.kl_divergence(y_true, y_pred, reduction: :mean)
+  end
+
+  def build_model(word_count, label_count) do
+      Axon.input({nil, @sequence_length, 1}, "inputs")
+       |> Axon.embedding(word_count, @sequence_length)
+       |> Axon.nx(fn t ->
+         t[[0..-1//1, -1]] # Dropping out the sized 1 column created from the embed
+       end)
+       |> Axon.spatial_dropout(rate: 0.1)
```

Comment:
I'm not sure this is a good idea---a better approach would be to use recurrent dropout, but Axon does not support that yet :)

Reply:
I like this feedback a lot, using spatial_dropout results in some strange things happening, I've been doing testing on this. So you propose recurrent dropout? I'll have to look into what it is and see what we can do with it. The spatial dropout I believe seems to cause issues after X epochs, I assume it's due to it dropping same channel across the tensors, which could be the feature map.